cdap-docs/admin-manual/source/installation/replication.rst

```diff
@@ -0,0 +1,256 @@
+- Enable replication from master to slave::
+
+    master_hbase_shell> add_peer '[slave-name]', '[slave-zookeeper-quorum]:/[slave-zk-node]'
+ 
+    # For example:
+    master_hbase_shell> add_peer 'hbase2', 'hbase2-master.example.com:2181:/hbase'
+
+- Enable replication from slave to master::
+
+    slave_hbase_shell> add_peer '[master-name]', '[master-zookeeper-quorum]:/[master-zk-node]'
+ 
+    # For example:
+    slave_hbase_shell> add_peer 'hbase1', 'hbase1-master.example.com:2181:/hbase'
+    
+- Confirm that HBase replication is working::
+
+    master_hbase_shell> create 'repltest', 'f'
+    
+    slave_hbase_shell> create 'repltest', 'f'
+    
+    master_hbase_shell> enable_table_replication 'repltest'
+    
+    slave_hbase_shell> alter 'repltest', { 'NAME' => 'f', 'REPLICATION_SCOPE' => 1 }
+    
+    master_hbase_shell> put 'repltest', 'masterrow', 'f:v1', 'v1'
+    
+    slave_hbase_shell> put 'repltest', 'slaverow', 'f:v1', 'v1'
+    
+    master_hbase_shell> scan 'repltest'
+    
+    slave_hbase_shell> scan 'repltest'
+
+HDFS
+----
+Set up HDFS replication using the solution provided by your distribution. HDFS does not
+have true replication, but it is usually achieved by scheduling regular ``distcp`` jobs.
+
+Hive
+----
+Set up replication for the database backing your Hive Metastore. Note that this will
+simply replicate the Hive metadata |---| which tables exist, table metadata, etc. |---|
+but not the data itself. It is assumed you will not be running Hive queries on the slave
+until after a manual failover occurs.
+
+For example, to setup MySQL 5.7 replication, follow the steps described at 
+`Setting Up Binary Log File Position Based Replication 
+<https://dev.mysql.com/doc/refman/5.7/en/replication-howto.html>`__.
+             
+Kafka
+-----
+Set up replication for the Kafka brokers you are using. *Kafka MirrorMaker* is the most
+common solution. See `Mirroring data between clusters 
+<https://kafka.apache.org/documentation.html#basic_ops_mirror_maker>`__ and
+`Kafka mirroring (MirrorMaker) 
+<https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=27846330>`__
+for additional information.
+
+.. _installation-replication-cdap-setup:
+
+CDAP Setup
+==========
+CDAP requires that you provide an extension that will perform HBase-related DDL operations
+on both clusters instead of only on a single cluster. To create the extension, you must
+implement the ``HBaseDDLExecutor`` class. Details on implementing this class, a sample
+implementation, and example files are available in the :ref:`Appendix: HBaseDDLExecutor
+<appendix-hbase-ddl-executor>`.
+
+CDAP must have :ref:`invalid transaction list pruning disabled 
+<installation-replication-disable-invalid-transaction-list-pruning>`, as this cannot be
+used with replication.
+
+.. highlight:: console
+
+To deploy your extension (once compiled and packaged as a JAR file, such as
+*my-extension.jar*), run these steps on **both** your master and slave clusters:
```

Comment:
`...your master and slave clusters. Following steps assume <cdap-home> is /opt/cdap`: